lib/faraday/response/json.rb

```diff
@@ -27,7 +29,11 @@ def process_response(env)
       end
 
       def parse(body)
-        ::JSON.parse(body, @parser_options || {}) unless body.strip.empty?
+        return if body.strip.empty?
+
+        decoder, method_name = @decoder_options
+
+        decoder.public_send(method_name, body, @parser_options)
```

Comment:
I see you removed the `|| {}` from `@parser_options`, but this can be `nil`.
It looks like `JSON.parse` is OK with that, but I tried running `Oj.load('{}', nil)` and I got an error as it expects an hash to be passed. That would make the documentation examples fail

Reply:
Whoops, my mistake, shaky hand :)